knimin/handlers/barcode_util.py

```diff
@@ -305,3 +312,48 @@ def post(self):
                     msgs=(gen_update_msg, email_msg, ag_update_msg,
                           project_msg),
                     currentuser=self.current_user)
+
+
+def update_ag_metadata(barcode, md):
+    """Push the sample's metadata to Qiita
+
+    This is specific for American Gut samples. We need to make sure the sample
+    is represented in Qiita so that it can be plated by the PlateMapper
+    interface.
+    """
+    sc, response = qiita_client.get('/api/v1/study/10317/samples')
+    if sc != 200:
+        return "Unable to get sample IDs from Qiita"
+
+    existing = set(json_decode(response.body))
+    if barcode in existing:
+        return "Metadata for %s already exists in Qiita" % barcode
+
+    # the db.pulldown call returns formatted metadata, so we actually need to
+    # reparse it unfortunately.
+    md = pd.read_csv(md, sep='\t', dtype=str)
```

Comment:
Unfortunately `dtype=str` is not sufficient to preserve string-only types, I suggest that you also add: `na_values=[], keep_default_na=False`.